client-data/tools/hand/hand.js

```diff
@@ -50,6 +64,108 @@
 			draw(msg);
 		}
 	}
+	var parentMathematics = els.find(el => el.getAttribute("class") === "MathElement");
+	if ((parentMathematics) && parentMathematics.tagName === "svg") {
+		target = parentMathematics;
+	}
+	return target ?? el;
+    }
+
+    function createSelectorRect() {
+	var shape = Tools.createSVGElement("rect");
+	shape.id = "selectionRect";
+	shape.x.baseVal.value = 0;
+	shape.y.baseVal.value = 0;
+	shape.width.baseVal.value = 0;
+	shape.height.baseVal.value = 0;
+	shape.setAttribute("stroke", "black");
+	shape.setAttribute("stroke-width", 3);
+	shape.setAttribute("fill", "none");
+	shape.setAttribute("stroke-dasharray", "5 5");
+	shape.setAttribute("opacity", 1);
+	Tools.svg.appendChild(shape);
+	return shape;
+    }
+
+    function startMovingElements(x, y, evt) {
+	evt.preventDefault();
+	selectorState = selectorStates.moving;
+	selected = {x: x, y: y};
+	// Some of the selected elements could have been deleted
+	selected_els = selected_els.filter(el=>{
+	    return Tools.svg.getElementById(el.id) !== null
+	});
+	translation_elements = selected_els.map(el => {
+	    let tmatrix = get_translate_matrix(el);
+	    return {x: tmatrix.e, y: tmatrix.f}
+	});
+	{
+	    let tmatrix = get_translate_matrix(selectionRect);
+	    selectionRectTranslation = {x: tmatrix.e, y: tmatrix.f};
+	}
+    }
+
+    function startSelector(x, y , evt) {
+	evt.preventDefault();
+	selected = {x: x, y: y};
+	selected_els= [];
+	selectorState = selectorStates.selecting;
+	selectionRect.x.baseVal.value = x;
+	selectionRect.y.baseVal.value = y;
+	selectionRect.width.baseVal.value = 0;
+	selectionRect.height.baseVal.value = 0;
+	selectionRect.style.display = "";
+	tmatrix = get_translate_matrix(selectionRect);
+	tmatrix.e = 0;
+	tmatrix.f = 0;
+    }
+
+
+    function calculateSelection() {
+	var scale = Tools.drawingArea.getCTM().a;
+	var selectionTBBox = selectionRect .transformedBBox(scale);
+	return Array.from(Tools.drawingArea.children).filter(el => {
```

Comment:
Isn't that going to just freeze the browser once there is more than just a few elements on the board ?
Did you try that on a board such as https://wbo.ophir.dev/boards/anonymous ?

Reply:
No I haven't tried it on such a full board. Is there any way that I could download or you could send the board so I can try it?

Reply:
Okay, I tested it on https://wbo.ophir.dev/boards/anonymous and it handles the board without a noticeable freeze in the browser.

Reply:
I did test it locally, and even though the framerate drops to some low values during selection, I agree that the overall performance is acceptable, and I think that the improvements are worth the cost.
So I apologize for my initial knee jerk reaction, and I'll try to review this in details and merge this week.

Thank you again.